x-pack/plugins/observability_solution/dataset_quality/public/hooks/use_dataset_quality_flyout.tsx

```diff
@@ -23,8 +23,12 @@ export const useDatasetQualityFlyout = () => {
   } = useSelector(service, (state) => state.context.flyout);
   const { timeRange } = useSelector(service, (state) => state.context.filters);
 
-  const dataStreamDetailsLoading = useSelector(service, (state) =>
-    state.matches('datasets.loaded.flyoutOpen.fetching')
+  const dataStreamDetailsLoading = useSelector(
+    service,
+    (state) =>
+      state.matches('datasets.loaded.flyoutOpen.fetching') ||
+      state.matches('flyout.initializing.dataStreamDetails.fetching') ||
+      state.matches('flyout.initializing.integrationDashboards.fetching')
```

Comment:
Wouldn't block the UI because of the integration dashboards, from what I see this loading state is being used to render `dataStreamDetails` and `integrationSummary`, but dashboards are only visible when the user expand the actions, why would we want to wait for the dashboards to be loaded? users might not even open the dashboards from the actions?